From 1f4db88c82daa9fdb532259a70bc47156394db96 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 15 Mar 2021 22:49:26 +0000
Subject: [PATCH] gnu: Add org-roam-server

* gnu/packages/emacs-xyz.scm: (emacs-org-roam-server): New variable
---
 gnu/packages/emacs-xyz.scm | 44 +++++++++++++++++++++++++++++++++++++-
 1 file changed, 43 insertions(+), 1 deletion(-)

diff --git a/gnu/packages/emacs-xyz.scm b/gnu/packages/emacs-xyz.scm
index dabac3c287..46757789e2 100644
--- a/gnu/packages/emacs-xyz.scm
+++ b/gnu/packages/emacs-xyz.scm
@@ -95,6 +95,7 @@
 ;;; Copyright © 2021 Alexey Abramov <levenson@mmer.org>
 ;;; Copyright © 2021 Xinglu Chen <public@yoctocell.xyz>
 ;;; Copyright © 2021 Stefan Reichör <stefan@xsteve.at>
+;;; Copyright © 2021 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -139,8 +140,9 @@
   #:use-module (gnu packages djvu)
   #:use-module (gnu packages ebook)
   #:use-module (gnu packages emacs)
-  #:use-module (gnu packages guile)
+  #:use-module (gnu packages freedesktop)
   #:use-module (gnu packages gtk)
+  #:use-module (gnu packages guile)
   #:use-module (gnu packages gnome)
   #:use-module (gnu packages haskell-apps)
   #:use-module (gnu packages ibus)
@@ -26443,6 +26445,46 @@ bibliographical notes in Org Roam directory via Helm BibTeX, Ivy BibTeX, or by
 opening Org Ref's @code{cite:} links.")
     (license license:gpl3+)))
 
+(define-public emacs-org-roam-server
+  (package
+    (name "emacs-org-roam-server")
+    (version "1.1.2")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/org-roam/org-roam-server")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1jp8mkqx1l3w166b16l2d5zsqjcc836bkclplgjk4laysb6msry8"))))
+    (build-system emacs-build-system)
+    (arguments
+     `(#:phases
+       (modify-phases %standard-phases
+         (add-after 'install 'install-assets
+           (lambda* (#:key outputs #:allow-other-keys)
+             (let* ((out (assoc-ref outputs "out"))
+                    (site-lisp (string-append out "/share/emacs/site-lisp"))
+                    (assets (string-append site-lisp "/assets")))
+               (mkdir-p assets)
+               (copy-recursively "assets" assets)
+               (copy-file "index.html" (string-append site-lisp "/index.html"))
+               #t))))))
+    (propagated-inputs
+     `(("emacs-f" ,emacs-f)
+       ("emacs-dash" ,emacs-dash)
+       ("emacs-simple-httpd" ,emacs-simple-httpd)
+       ("emacs-org-roam" ,emacs-org-roam)
+       ("emacs-s" ,emacs-s)
+       ("xdg-utils" ,xdg-utils))) ; for xdg-mime to load org-roam-protocol
+    (home-page "https://github.com/org-roam/org-roam-server")
+    (synopsis "Web Application to Visualize the Org-Roam Database")
+    (description "Web Application to Visualize the Org-Roam Database")
+    ;; jquery - MIT, vis-network - Apache, Bootstrap - MIT,
+    ;; orgcss - MIT, select2 - MIT, bootstrap-toggle - MIT
+    (license (list license:expat license:asl2.0))))
+
 (define-public emacs-org-generate
   (package
     (name "emacs-org-generate")
-- 
2.29.2

